websites/S/Steam/presence.ts

```diff
@@ -23,314 +19,471 @@ presence.on("UpdateData", async () => {
+				.querySelector<HTMLTitleElement>("head > title")
+				.textContent.replace("Steam Franchise: ", " ");
+			presenceData.startTimestamp = browsingTimestamp;
+			presenceData.buttons = [
+				{
+					label: "View the page",
+					url: document.location.href,
+				},
+			];
+			presenceData.smallImageKey = "lg";
+			presenceData.largeImageKey = document.querySelector<HTMLMetaElement>(
+				"head > meta[property='og:image']"
+			).content;
+		} else if (document.location.pathname.includes("/curator")) {
+			presenceData.state = "Viewing a Steam Curator page";
+			presenceData.startTimestamp = browsingTimestamp;
+		} else if (document.location.pathname.includes("/specials")) {
+			presenceData.state = "Viewing special offers";
+			presenceData.startTimestamp = browsingTimestamp;
+		} else if (document.location.pathname.includes("/charts")) {
+			presenceData.startTimestamp = browsingTimestamp;
+			presenceData.details = "Steam Store Charts";
+			if (document.location.pathname.includes("topselling"))
+				presenceData.state = "Top 100 Selling Games";
+			else if (document.location.pathname.includes("mostplayed"))
+				presenceData.state = "Top 100 Played Games";
+			else if (document.location.pathname.includes("topsellers")) {
+				const parted = document.location.href.split("/");
+				presenceData.state = `Top Sellers for ${parted[parted.length - 1]}`;
+			} else if (document.location.pathname.includes("topnewreleases")) {
+				const parted = document.location.href.split("/");
+				presenceData.state = `Top new releases for${parted[parted.length - 1]
+					.replace("top_", " ")
+					.replace("_", " ")}`;
+			} else
+				presenceData.state = "Top selling and top played games across Steam";
 		} else if (document.location.pathname.includes("/stats")) {
 			presenceData.state = "Steam & Games Stats.";
-
 			presenceData.startTimestamp = browsingTimestamp;
 		} else if (document.location.pathname.includes("/search")) {
-			presenceData.state = `Searching for ${new URL(
-				document.location.href
-			).searchParams.get("term")}`;
-
 			presenceData.startTimestamp = browsingTimestamp;
-
-			presenceData.smallImageKey = "search";
+			const SeachTerm = new URL(document.location.href).searchParams.get(
+				"term"
+			);
+			if (SeachTerm === null) presenceData.state = "Searching...";
+			else {
+				presenceData.state = `Searching for ${new URL(
+					document.location.href
+				).searchParams.get("term")}`;
+				presenceData.smallImageKey = "search";
+			}
 		}
+	} else if (document.location.hostname === "help.steampowered.com") {
+		presenceData.details = "Steam Support";
+		presenceData.startTimestamp = browsingTimestamp;
+		presenceData.state = document
+			.querySelector<HTMLTitleElement>("head > title")
+			.textContent.replace("Steam Support", " ")
+			.replace(" - ", " ");
+		presenceData.largeImageKey = document.querySelector<HTMLMetaElement>(
+			"head > meta[property='og:image']"
+		).content;
 	}
 
 	presence.setActivity(presenceData);
```

Comment:
To alleviate any problems from occurring, I suggest turning this into an if statement, like so:

```
if (presenceData.details) presence.setActivity(presenceData);
else presence.setActivity();
```

Reply:
okay